feat: add `originalResponse` to be available in the templating

Mohsin-Ul-Islam opened the pull request:
<!-- Please describe your pull request here. -->

## References

- Implements functionality for #2359

<!-- References to relevant GitHub issues and pull requests, esp. upstream and downstream changes -->

## Submitter checklist

- [ ] Recommended: Join [WireMock Slack](https://slack.wiremock.org/) to get any help in `#help-contributing` or a project-specific channel like `#wiremock-java`
- [x] The PR request is well described and justified, including the body and the references
- [ ] The PR title represents the desired changelog entry
- [x] The repository's code style is followed (see the contributing guide)
- [x] Test coverage that demonstrates that the change works as expected
- [ ] For new features, there's necessary documentation in this pull request or in a subsequent PR to [wiremock.org](https://github.com/wiremock/wiremock.org)

<!--
Put an `x` into the [ ] to show you have filled the information.
The template comes from https://github.com/wiremock/.github/blob/main/.github/pull_request_template.md 
You can override it by creating .github/pull_request_template.md in your own repository 
-->

leeturner commented on src/main/java/com/github/tomakehurst/wiremock/extension/responsetemplating/ResponseTemplateModel.java:
```diff
@@ -0,0 +1,39 @@
+  }
+
+  public String getBody() {
+    return body.asString();
```
Hi, many thanks for your PR.  I think we will need to take a look at this and how it is going to work when the `Body` isn't a string.   The body can be returned as base64 which we will need to factor in here I think.  At least allow the other body representations to be returned from the ResponseTemplateModel

Mohsin-Ul-Islam replied:
makes sense

leeturner commented on src/main/java/com/github/tomakehurst/wiremock/extension/responsetemplating/ResponseTemplateModel.java:
```diff
@@ -0,0 +1,39 @@
+  private final Body body;
+  private final Map<String, ListOrSingle<String>> headers;
+
+  public ResponseTemplateModel(Map<String, ListOrSingle<String>> headers, Body body) {
```
I was just wondering if there is any value to adding the response status here for a more complete response model?

Mohsin-Ul-Islam replied:
Yes, I think it is better to have response status in the model as well.

leeturner commented:
Many thanks for the PR.  I have added a few comments for you to take a look at.  Also, could we add a new test to validate the headers are transferred to the model OK and maybe what happens when there are no headers set in the response etc

Mohsin-Ul-Islam commented:
> Many thanks for the PR. I have added a few comments for you to take a look at. Also, could we add a new test to validate the headers are transferred to the model OK and maybe what happens when there are no headers set in the response etc

@leeturner I have updated the test to assert for headers, status & body. Please re-review.

Thanks

Mohsin-Ul-Islam commented:
> Also, if you have time, it would be great if you could add a PR to the docs site documenting the new addition to the webhook page - https://github.com/wiremock/wiremock.org/blob/main/_docs/webhooks-and-callbacks.md

@leeturner here is the PR containing documentation about this feature: https://github.com/wiremock/wiremock.org/pull/334

tomakehurst commented on src/main/java/com/github/tomakehurst/wiremock/extension/responsetemplating/ResponseTemplateModel.java:
```diff
@@ -0,0 +1,45 @@
+  }
+
+  public String getBody() {
+    return body.asBase64();
```
I don't think this is what we want to expose to the templates.

The best approach would be to return the string representation when it's a textual body and null otherwise IMO.

bernalvarela commented:
Hello, this feature looks really interesting.

It seems that this PR completes the work from the previously merged one: https://github.com/wiremock/wiremock/issues/2359.

Is there anything that can be done to unblock this PR and get it merged?

Thank you all for your work and best regards!!

Mohsin-Ul-Islam commented:
@leeturner @tomakehurst can we take this forward. Thanks for your support.
